generic_config_updater/generic_updater.py

```diff
@@ -0,0 +1,469 @@
+        return FormatConverter.to_serialized(data)
+
+    def get_sonic_yang_as_json(self):
+        config_db_json = self.get_config_db_as_json()
+        return self.convert_config_db_to_sonic_yang(config_db_json)
+
+    def convert_config_db_to_sonic_yang(self, config_db_as_json):
+        sy = sonic_yang.SonicYang(self.yang_dir)
+        sy.loadYangModel()
+
+        # Crop config_db tables that do not have sonic yang models
+        cropped_config_db_as_json = self.crop_tables_without_yang(config_db_as_json)
+
+        sonic_yang_as_json = dict()
+
+        sy._xlateConfigDBtoYang(cropped_config_db_as_json, sonic_yang_as_json)
+
+        return sonic_yang_as_json
+
+    def convert_sonic_yang_to_config_db(self, sonic_yang_as_json):
+        sy = sonic_yang.SonicYang(self.yang_dir)
+        sy.loadYangModel()
+
+        # replace container of the format 'module:table' with just 'table'
+        new_sonic_yang_json = {}
+        for module_top in sonic_yang_as_json.keys():
+            new_sonic_yang_json[module_top] = {}
+            for container in sonic_yang_as_json[module_top].keys():
+                table = container.split(':')[1]
+                new_sonic_yang_json[module_top][table] = sonic_yang_as_json[module_top][container]
+
+        config_db_as_json = dict()
+        sy.xlateJson = new_sonic_yang_json
+        sy.revXlateJson = config_db_as_json
+        sy._revXlateYangtoConfigDB(new_sonic_yang_json, config_db_as_json)
+
+        return config_db_as_json
+
+    def validate_sonic_yang_config(self, sonic_yang_as_json):
+        config_db_as_json = self.convert_sonic_yang_to_config_db(sonic_yang_as_json)
+
+        sy = sonic_yang.SonicYang(self.yang_dir)
+        sy.loadYangModel()
+
+        try:
+            sy.loadData(config_db_as_json)
+
+            sy.validate_data_tree()
+            return True
+        except Exception as ex:
+            return False
+
+    def crop_tables_without_yang(self, config_db_as_json):
+        sy = sonic_yang.SonicYang(self.yang_dir)
+        sy.loadYangModel()
+
+        sy.jIn = copy.deepcopy(config_db_as_json)
+
+        sy.tablesWithOutYang = dict()
+
+        sy._cropConfigDB()
+
+        return sy.jIn
+
+    def __create_and_connect_config_db(self):
+        if self.default_config_db_connector != None:
+            return self.default_config_db_connector
+
+        db_kwargs = dict()
+        config_db = ConfigDBConnector(**db_kwargs)
+        config_db.connect()
+        return config_db
+
+class DryRunConfigWrapper(ConfigWrapper):
+    # TODO: implement DryRunConfigWrapper
+    # This class will simulate all read/write operations to ConfigDB on a virtual storage unit.
+    pass
+
+class PatchWrapper:
+    def __init__(self, config_wrapper = ConfigWrapper()):
+        self.config_wrapper = config_wrapper
+
+    def validate_config_db_patch(self, patch):
+        config_db = {}
+        for operation in patch:
+            tokens = operation['path'].split('/')[1:]
```

Comment:
>split [](http://example.com/codeflow?start=39&length=5)

If no `/` in the string, index operator will throw

Reply:
looks OK
```python
operation = {}
operation['path']=''
tokens = operation['path'].split('/')[1:]
print(tokens)
[]
```

Reply:
That for testing! you are right.